src/Backend/App.vala

```diff
@@ -111,7 +122,15 @@ public class Slingshot.Backend.App : Object {
                     break;
                 case AppType.APP:
                     launched (this); // Emit launched signal
-                    new DesktopAppInfo (desktop_id).launch (null, null);
+
+                    var app_info = new DesktopAppInfo (desktop_id);
+                    bool use_default_gpu = !app_info.get_boolean ("PrefersNonDefaultGPU");
```

Comment:
I think I'd prefer to see this boolean as a field on the class that's set at construct time rather than looking up the desktop file again each time.